### ISSUE TITLE
CreateSharedLinkWithSettingsAsync not working

### ISSUE THREAD

**Issue by cylonmia** [dropbox/dropbox-sdk-dotnet#254]
It's working on version 6.10.1
stopped working on v 6.10.2
still throws error on v 6.10.3

**To Reproduce**
var link = await dbx.Sharing.CreateSharedLinkWithSettingsAsync(result.Metadata.PathDisplay);

**Expected Behavior**
link has a value, it's the shared link

**Actual Behavior**
throws exception "Operation is not valid due to the current state of the object"
stack trace takes me back to Dropbox.api.sharing.routes.sharinguserroutes

**Screenshots**
No.

**Versions**
* What version of the SDK are you using?
* 6.10.2 or 6.10.3
* What version of the language are you using?
* C#
* What platform are you using? (if applicable)
* Visual Studio

**Additional context**
sorry, I'm on a deadline and this error has made me waste too much time.
but it's easy to repro, just 
var dbx = new DropboxClient(dbtoken);
var folderArg = new CreateFolderArg(targetPath, true);
var result = await dbx.Files.CreateFolderV2Async(folderArg);
var link = await dbx.Sharing.CreateSharedLinkWithSettingsAsync(result.Metadata.PathDisplay);

**Comment by greg-db** [comment]
Thanks for the report! We'll look into it.

**Comment by sabihismail** [comment]
I get the same issue with GetSharedLinkFileAsync. Through curl it works as expected but when I submit the same request with no path it throws the same issue. The CURL request is 
```
curl -X POST https://content.dropboxapi.com/2/sharing/get_shared_link_file \
    --header "Authorization: Bearer <token>" \
    --header "Dropbox-API-Arg: {\"url\": \"https://www.dropbox.com/s/<some_id>/file.txt?dl=0\"}"
```
    
which works as expected, but when I try 

```
GetSharedLinkFileAsync("https://www.dropbox.com/s/<some_id>/file.txt?dl=0")
``` 

it throws the same `Operation is not valid due to the current state of the object.` exception.

**Comment by greg-db** [comment]
Thanks @sabihismail . Supplying the shared link without path when it's for a file is supposed to be valid. This appears to be the same issue as with `CreateSharedLinkWithSettingsAsync`. I'll follow up here once I have an update on this.

**Comment by Flexicon** [comment]
The obsolete method `CreateSharedLinkAsync` as well as just sending a curl request both work - however I can confirm that `CreateSharedLinkWithSettingsAsync` does not work.

---

This works:

```
var sharedLink = await client.Sharing.CreateSharedLinkAsync(new CreateSharedLinkArg(metadata.PathLower));
```

This fails:

```
var sharedLink = await client.Sharing.CreateSharedLinkWithSettingsAsync(new CreateSharedLinkWithSettingsArg(metadata.PathLower));
```

**Comment by greg-db** [comment]
This should be fixed in the latest version now, v6.11.1.
